concordium-consensus/src/Concordium/ImportExport.hs

```diff
@@ -367,6 +381,85 @@ exportDatabaseV3 dbDir outDir chunkSize = do
             return exportError
         Nothing -> return exportError
 
+-- |Return the last exported genesis hash or the provided one.
+exportedGenHashOr :: BlockHash -> BlockIndex -> BlockHash
+exportedGenHashOr bh blockIndex = case blockIndex of
+    -- nothing was previously exported for the
+    -- section corresponding to the current
+    -- genesis index.
+    Empty -> bh
+    -- blocks were previously exported for the
+    -- current genesis index, so use the genesis
+    -- block hash that was previously exported
+    -- for that section and compare it with that
+    -- of the database.
+    _ :|> (gh, _) -> gh
+
+-- |Export blocks from a 'ConsensusV1' database.
+exportConsensusV1Blocks ::
+    forall pv m.
+    ( IsProtocolVersion pv,
+      MonadIO m,
+      KonsensusV1.MonadTreeStateStore m
+    ) =>
+    -- |Export path.
+    FilePath ->
+    -- |Chunk size
+    Word64 ->
+    -- |The genesis index.
+    GenesisIndex ->
+    -- |Height to start export from.
+    BlockHeight ->
+    -- |The block index of the previous export.
+    BlockIndex ->
+    -- |Last written chunk in previous export
+    Maybe String ->
+    -- |Returns a @Bool@ which indicates whether anything went wrong,
+    -- i.e. it is 'True' if an error occured and otherwise 'False,
+    -- and the resulting 'BlockIndex'.
+    m (Bool, BlockIndex)
+exportConsensusV1Blocks outDir chunkSize genIndex startHeight blockIndex lastWrittenChunkM = do
+    KonsensusV1.lookupFirstBlock >>= \case
+        Nothing -> do
+            liftIO . putStrLn $ "Could not read from database."
```

Comment:
Could we not instead use the logging infrastructure here? It's a bit odd we have runSilentLogger and then here we use println.

Reply:
Introduced a logger for all output.